app/export.go

```diff
@@ -62,6 +65,62 @@ var exportablePreferences = map[ComparablePreference]string{{
 }: "EmailInterval",
 }
 
+func (a *App) GetID() string {
+	return a.ID
+}
+func (a *App) SetID(id string) {
+	a.ID = id
+}
+
+func exportLogFile(attachment AttachmentImportData, exportFilename string) (*LogFormat, *model.AppError) {
+	createdAtTime, err := strconv.ParseInt(strconv.FormatInt(attachment.CreateAt, 10), 10, 64)
+	if err != nil {
+		return nil, model.NewAppError("parseTime", "Failed to parse createdAt time", nil, err.Error(), http.StatusInternalServerError)
+	}
+	updatedAtTime, err := strconv.ParseInt(strconv.FormatInt(attachment.UpdateAt, 10), 10, 64)
+	if err != nil {
+		return nil, model.NewAppError("parseTime", "Failed to parse updatedAt time", nil, err.Error(), http.StatusInternalServerError)
+	}
+	data := LogFormat{
+		Id:            attachment.Id,
+		CreatorId:     attachment.CreatorId,
+		PostId:        attachment.PostId,
+		ChannelId:     attachment.ChannelId,
+		CreateAt:      time.UnixMilli(createdAtTime),
+		UpdateAt:      time.UnixMilli(updatedAtTime),
+		Path:          *attachment.Path,
+		ThumbnailPath: attachment.ThumbnailPath,
+		PreviewPath:   attachment.PreviewPath,
+		Name:          attachment.Name,
+		Extension:     attachment.Extension,
+		Size:          attachment.Size,
+		MimeType:      attachment.MimeType,
+		Width:         attachment.Width,
+		Height:        attachment.Height,
+	}
+	return &data, nil
+}
+
+func (a *App) ExportAttachments(attachment AttachmentImportData, exportFilename string, missingAttachments *[]LogFormat, outPath string, zipWr *zip.Writer) *model.AppError {
```

Comment:
```suggestion
func (a *App) ExportAttachments(attachment AttachmentImportData, exportFilename string, missingAttachments []*LogFormat, outPath string, zipWr *zip.Writer) *model.AppError {
```
Also, no use of exportFilename.

Reply:
I think that in this we should make whole array as a pointer. Because we are appending something in this array, like if we do not make array as a pointer then we will get size of our array as zero in `BulkExport` function